### PR TITLE
Pull-card: Add page showing them in many states

### PR DESCRIPTION
**Pull request by danielbeardsley**
This is mainly for QAing and testing out how pieces of our app look like in many different states.

image

### CODE REVIEW

**Comment by danielbeardsley** [comment]
un_dev_block :+1:

**Comment by deltuh-vee** [comment]
@danielbeardsley 
I noticed the Dev Blocked card and Milestones cards don't have their typical icon flags. Is that intentional?
CleanShot 2022-03-14 at 13 56 17

**Comment by danielbeardsley** [comment]
> I noticed the Dev Blocked card and Milestones cards don't have their typical icon flags. Is that intentional?

See #247

**Comment by deltuh-vee** [comment]
QA 🎬 
pull-card-demo page has multiple cards in various states.

**Comment by danielbeardsley** [comment]
QA need not be repeated.  QA :+1:

**Comment by mlahargou** [comment]
CR 📱

**Comment by cdcline** [comment]
CR 👍

File: frontend/test/named-pulls.ts [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+      status: {
+         allQA: [activeQA],
+         allCR: [activeCR],
+      },
+   }),
+   pullData({
+      title: "1 CR and 1 QA, but no requirements",
+      cr_req: 0,
+      qa_req: 0,
+   }),
```
Comment (andyg0808) [inline frontend/test/named-pulls.ts:60]: This doesn't look like it's actually got any CR or QA on it despite the title, if I'm understanding the way this works correctly.

File: frontend/test/named-pulls.ts [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+      title: "2 CR and 2 QA but 2,1 is required",
+      cr_req: 2,
+      qa_req: 1,
+      status: {
+         allQA: [activeQA, activeQA2],
+         allCR: [activeCR, activeCR2],
+      },
+   }),
+   pullData({
+      title: "2 active and 1 out of date CR where only 2 are required. Same with QA",
```
Comment (andyg0808) [inline frontend/test/named-pulls.ts:83]: Technically, QA only requires one, and this seems to imply it requires two.

File: frontend/test/named-pulls.ts [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+
+export const MyOwn = <PullData[]> [
+   pullData({
+      title: "Pull Created By Me",
+      user: {
+         login: getUser(),
+      }
+   }),
+   pullData({
+      title: "With an up-to-date CR and a QA by Me",
```
Comment (andyg0808) [inline frontend/test/named-pulls.ts:218]:
```suggestion
      title: "With an up-to-date CR and QA by Me",
```

File: frontend/test/named-pulls.ts [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+      status: {
+         allQA: [sig(SignatureType.QA, true, getUser())],
+         allCR: [sig(SignatureType.CR, true, getUser())],
+      },
+   }),
+   pullData({
+      title: "With an out-of-date CR and a QA by Me",
```
Comment (andyg0808) [inline frontend/test/named-pulls.ts:225]:
```suggestion
      title: "With an out-of-date CR and QA by Me",
```

**Review by andyg0808 — APPROVED** [review]

File: frontend/test/named-pulls.ts (hunk 1 above)
Comment (danielbeardsley) [reply frontend/test/named-pulls.ts:60]: Derpy derp.

**Comment by danielbeardsley** [comment]
I'll fix these in #247

File: frontend/test/named-pulls.ts (hunk 4 above)
Comment (danielbeardsley) [reply frontend/test/named-pulls.ts:225]: Fixed in the other pull

File: frontend/test/named-pulls.ts (hunk 3 above)
Comment (danielbeardsley) [reply frontend/test/named-pulls.ts:218]: I think both are valid. `I ate an apple and an orange`  or  `I ate an apple and orange` both seem sensible with the former sounding more natural to me.

File: frontend/test/named-pulls.ts (hunk 2 above)
Comment (danielbeardsley) [reply frontend/test/named-pulls.ts:83]: Fixed.

File: frontend/test/named-pulls.ts (hunk 3 above)
Comment (andyg0808) [reply frontend/test/named-pulls.ts:218]: Totally. To my ear, at least, though, the parentheses go like
```
With (an up-to-date CR) and (a QA by Me)
```
whereas 
```
With (an up-to-date CR and QA) by Me
```

I can't say for sure, though.

**Pull request merged**